### DISCUSSION TITLE
mini.files: file preview width

### DISCUSSION THREAD

**Discussion by lukebarton** [nvim-mini/mini.nvim#896] [Ideas]
Having a wide preview is awkward when you're navigating directories, but desirable when previewing files.

It'd nice to preview files wide, but directories narrow, so the interface is cleaner. For example, I'd have everything set to 30, which'd give a nice clean browsing experience, except for file previews, which I'd make 90.

**Comment by echasnovski** [comment]
You can customize [`width_focus`, `width_nofocus`, and `width_preview`](https://github.com/echasnovski/mini.nvim/blob/eb2dd6d187e1ab5fefec66f0d37b1a3dc8633d17/doc/mini-files.txt#L541-L546).

**Reply by lukebarton** [reply]
Yes, but directory previews use `width_preview`, right?

**Reply by lukebarton** [reply]
What I was, perhaps poorly trying to explain is, that I'd prefer it if I could make all the windows fixed width, except for _file_ previews which'd be larger. When you're previewing directories at a larger width, especially if you're navigating back in the tree, makes the display kind crazy and hard to follow.

Ideally, I'd be able to configure _file_ previews separately from directory previews, so the navigation of directories is consistent, but the preview of files is larger.

**Reply by echasnovski** [reply]
Ah, it is more about adding different width for file preview specifically. Sorry, missed that.
I'll think about it, but taking into account that 'mini.files' already has quite an elaborate config options, I am hesitant about adding a new one.

**Reply by lukebarton** [reply]
https://github.com/echasnovski/mini.nvim/assets/98427/69ba5337-1762-4c12-8a6d-80d9de55d33c

There's no need for a huge directory preview, and the ux is a bit weird ^, but a small file preview is limited in use

**Comment by derrekcoleman** [comment]
tysm for mini.files, @echasnovski! Throwing a +1 on this closed idea; I wish I could have different `width_preview` depending on whether it was a directory or a file

**Comment by julius-rennmann** [comment]
@echasnovski Hi, I am also interested in this feature. Would you consider merging it, if I opened a PR?

**Reply by echasnovski** [reply]
Thanks for the suggestion! Not currently, no. The [earlier reasons](https://github.com/echasnovski/mini.nvim/discussions/896#discussioncomment-9509628) still apply. But I'll keep this in mind in the near future, maybe it'll change.

**Reply by julius-rennmann** [reply]
All clear, did it anyways without opening a PR for fun and personal usage. No breaking changes.
For anyone interested: https://github.com/julius-rennmann/mini.nvim

**Comment by Myrrrca** [comment]
hi @echasnovski im also interested in this feature, maybe you could merge some PR if i open it?

**Reply by echasnovski** [reply]
Hi! I am afraid, no. Not at the moment. The [earlier reasons](https://github.com/nvim-mini/mini.nvim/discussions/896#discussioncomment-14222752) still apply.
